docs/DOCUMENTATION_GUIDELINES.md

```diff
@@ -0,0 +1,227 @@
+# Substrate Documentation Guidelines
+
+This document is only focused on documenting parts of substrate that relates to its external API. The list of such crates can be found in [CODEOWNERS](./CODEOWNERS). Search for the crates that are auto-assigned to a team called `docs-audit`.
+
+These are crates that are often used by external developers and deserve a better documents. Consequently, most of this revolves around FRAME-development.
+
+- [Substrate Documentation Guidelines](#substrate-documentation-guidelines)
+  - [General/Non-Pallet Crates](#generalnon-pallet-crates)
+    - [What to Document?](#what-to-document)
+      - [Rust Docs vs. Code Comments](#rust-docs-vs-code-comments)
+    - [How to Document?](#how-to-document)
+    - [TLDR](#tldr)
+    - [Other Guidelines](#other-guidelines)
+      - [Document Through Code](#document-through-code)
+      - [Formatting Matters](#formatting-matters)
+  - [Pallet Crates](#pallet-crates)
+    - [Top Level Pallet Docs (`lib.rs`)](#top-level-pallet-docs-librs)
+    - [Dispatchable](#dispatchable)
+    - [Storage Items](#storage-items)
+
+
+## General/Non-Pallet Crates
+
+First, consider the case for all such crates, except for those that are pallets.
+
+### What to Document?
+
+The first question is, what should you document? Use the following filter:
+
+1. Within the set of crates above,
+2. Any `pub` item within them needs to be documented. If it is not `pub`, it won't show up in the rust-docs, and won't matter.
+    * Within `pub` items, sometimes they are only `pub` in order to be used by another internal crate, and you can foresee that this will not be used by anyone else other than you. These need **not** be documented thoroughly, and are left to your discretion to identify.
```

Comment:
```suggestion
    * There are some `pub` items that are only `pub` in order to be accessible for another internal crate. These should still be documented although you do not need to be as thorough as you are with the `pub` items that are for external developer use.
```

Reply:
Should we ask to add a comment like "//no-doc" to establish that the author intentionally didn't add the docs?

Reply:
@hummusonrails I am a bit hesitant about your suggestions that seem paraphrasing to me. Do you have strong opinions that the original text is wrong or poorly worded, or are you adding optional suggestions? I accepted most, but might close some. Please let me know if you prefer otherwise.

Reply:
@kianenigma In this one, I attempted to make it more neutral and formal, whereas your version was more informal and personal in tone. I'm of the opinion that documentation should lean more into the former. That's why I made that suggestion. Hope that helps clarify!